pkg/kapp/diff/change_set_with_versioned_rs_test.go

```diff
@@ -0,0 +1,195 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package diff
+
+import (
+	"testing"
+
+	ctlres "github.com/k14s/kapp/pkg/kapp/resources"
+)
+
+func TestChangeSet_ExistingVersioned_NewNonVersioned_Resource(t *testing.T) {
+	newRs, _ := ctlres.NewResourcesFromBytes([]byte(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret
+`))
+
+	existingRes, _ := ctlres.NewResourcesFromBytes([]byte(`
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-ver-1
+  annotations:
+    kapp.k14s.io/versioned: ""
+`))
+
+	changeSetWithVerRes := NewChangeSetWithVersionedRs(existingRes, newRs, nil, ChangeSetOpts{}, ChangeFactory{})
+
+	changes, err := changeSetWithVerRes.Calculate()
+
+	if err != nil {
+		t.Fatalf("Expected non-error")
+	}
+
+	if changes[0].Op() != ChangeOpDelete {
+		t.Fatalf("Expected to get deleted: actual >>>%s<<< vs expected >>>%s<<<", changes[0].Op(), ChangeOpDelete)
+	}
+
+	if changes[1].Op() != ChangeOpAdd {
+		t.Fatalf("Expected to get added: actual >>>%s<<< vs expected >>>%s<<<", changes[1].Op(), ChangeOpAdd)
+	}
+
+	actualDiff1 := changes[0].ConfigurableTextDiff().Full().FullString()
+
+	expectedDiff1 := `  0,  0 - apiVersion: v1
+  1,  0 - kind: Secret
+  2,  0 - metadata:
+  3,  0 -   annotations:
+  4,  0 -     kapp.k14s.io/versioned: ""
+  5,  0 -   name: secret-ver-1
+  6,  0 - 
+`
+	if actualDiff1 != expectedDiff1 {
+		t.Fatalf("Expected diff to match: actual >>>%s<<< vs expected >>>%s<<< %d %d", actualDiff1, expectedDiff1, len(actualDiff1), len(expectedDiff1))
+	}
```

Comment:
this might be a good thing to extract into a little function so that it's not making test noisy. e.g.

```go
checkChangeDiff(t, changes[0], `...
...
`)
```